### ISSUE TITLE
ngrok

### ISSUE THREAD

**Issue by yungloi** [ngrok/ngrok#54]
Session Status                reconnecting (failed to dial ngrok server with address "connect.ngrok-agent.com:443": dial tcp

**Comment by russorat** [comment]
@yungloi thanks for the issue. tough to help without much info, but this usually means some software id blocking access to ngrok on your network. It might be some firewall software running.

**Comment by russorat** [comment]
closing this since there was no response. please open an issue with more details if you are still having trouble.

**Comment by jonkwheeler** [comment]
I just lost my mind on this one. It's fix now, and I have no idea why.

For context, I'm on Eero router, which uses 192.168.4.1 as a Gateway IP Address. I was consistantly getting this...

```
ngrok http --url jon-MY_DOMAIN.ngrok.app 8080 --log=stdout
INFO[10-31|17:34:52] no configuration paths supplied
INFO[10-31|17:34:52] using configuration at default config path path="/Users/jonwheeler/Library/Application Support/ngrok/ngrok.yml"
INFO[10-31|17:34:52] open config file                         path="/Users/jonwheeler/Library/Application Support/ngrok/ngrok.yml" err=nil
t=2024-10-31T17:34:52-0400 lvl=info msg="starting web service" obj=web addr=127.0.0.1:4040 allow_hosts=[]
t=2024-10-31T17:34:53-0400 lvl=eror msg="failed to reconnect session" obj=tunnels.session err="failed to dial ngrok server with address \"connect.ngrok-agent.com:443\": dial tcp 192.168.4.1:443: connect: connection refused"
```

I ended up in my Eero's App interface and rebooted a few times. Didn't change anything. And now it's magically working again?

However, I still get this warning.

```
ngrok diagnose
Testing ngrok connectivity...

Internet Connectivity
  Name Resolution                           [ OK ]
  TCP                                       [ OK ]
  TLS                                       [ OK ]
Localhost Connectivity
  Name Resolution                           [ OK ]
Ngrok Connectivity - Region: Auto (lowest latency)
  Name Resolution                           [ OK ]
  TCP                                     [ WARN ]
  TLS                                       [ OK ]
  Tunnel Protocol                           [ OK ]
Successfully established ngrok connection! (region: 'us', latency: 52.434709ms)

Warnings and non-fatal errors encountered during diagnostics:

* Diagnostics
  * Connectivity
    * connect.ngrok-agent.com (192.168.4.1:443)
      - Err: Failed to establish TCP connection to 192.168.4.1 with error: dial
             tcp 192.168.4.1:443: connect: connection refused
             (NGROK_ERR_8002)
```

**Comment by russorat** [comment]
@jonkwheeler thanks for the comment. That seems like a new thing. Connect.ngrok-agent.com shouldn't be resolving to an internal IP. Do you mind opening a new issue for this?

**Comment by kaganjd** [comment]
hey @jonkwheeler, huge thanks for the tip about Eero here- we've gotten a few similar reports over the last day or so and have been struggling to track down a root cause. Could you please try adding our agent connect hostname to the router's allow list? I think you can do this by going to Eero app > Discover > Eero Plus/Secure > Block & Allow Sites and allow `connect.ngrok-agent.com`. Please let us know how it goes!

**Comment by gsafcik** [comment]
Thank you @jonkwheeler and @kaganjd! I was also having this problem and on an eero router. It used to work but lately has not. I tried everything from reinstalling the agent to changing nameservers on my Mac. I can confirm adding it to the allow list works.

I added `connect.ngrok-agent.com` to the Allow list and it connected right away. I couldn't find any reference as to it being blocked from my Mac (no firewalls on) and couldn't find it blocked within eero (although I don't think it gives you specifics on what is blocked).
